### ISSUE TITLE
Tutorial: please clarify use of spread properties

### ISSUE THREAD

**Issue by xpe** [facebook/react#8177]
This part of the tutorial is confusing:

> Data change without mutation

```
var player = {score: 1}
player = {...player, score: 2} // new object not mutated {score: 2}
```

> The end result is the same but by not mutating (or changing the underlying data) directly we now have an added benefit that can help us increase component and overall application performance.

It does not appear to match up with either:

* JSX's spread operator
* the JS Spread operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator

The closest match I can find is here: https://github.com/sebmarkbage/ecmascript-rest-spread/blob/master/Spread.md

> Shallow Clone (excluding prototype)

> `let aClone = { ...a };`

**Comment by rthor** [comment]
This is a part of the [Object Rest/Spread Properties](https://github.com/sebmarkbage/ecmascript-rest-spread) proposal for ECMAScript that is not yet finalized; currently at stage 3. In the example given, the operator copies the key/value pairs from `player` onto a new object; then overrides the `score` value. The cloning process can be rewritten to use the [`Object.assign`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) method, which is currently in the spec (but requires a polyfill): 
```js
const player = {score: 1}
const playerCopy = Object.assign({}, player, {score: 2}) // new object, not mutated {score: 2}
```

----

As for updating the tutorial, I think that it, like the docs, should aim to use standards over proposed language features. However, if proposed features are popular, like Object Spread, then it can be added as an afterthought but indicating clearly that it needs a polyfill/transpiler/etc. In this case however, it might suffice to link to [the docs](https://facebook.github.io/react/docs/optimizing-performance.html#the-power-of-not-mutating-data) since this is covered quite well there.

**Comment by gaearon** [comment]
This is because the tutorial was adapter from Facebook's internal tutorial, and we have this feature enabled by default. This needs to be fixed in the docs.

@rthor Would you like to submit a doc PR fixing this?

**Comment by rthor** [comment]
Sure, I'll write it up later this week.

**Comment by rthor** [comment]
Seems to have been resolved in #8273.
